velox/dwio/dwrf/writer/ColumnWriter.cpp

```diff
@@ -1395,13 +1408,17 @@ uint64_t FloatColumnWriter<T>::write(
         writer.add(val);
         statsBuilder.addValues(val);
       };
-
-      for (auto& pos : ranges) {
-        if (bits::isBitNull(nulls, pos)) {
-          ++nullCount;
-        } else {
-          processRow(pos);
+      try {
+        for (auto& pos : ranges) {
+          if (bits::isBitNull(nulls, pos)) {
+            ++nullCount;
+          } else {
+            processRow(pos);
+          }
         }
+      } catch (std::exception& e) {
```

Comment:
Would it be better to use a folly::guard?

Reply:
Yeah, it would better. thanks for offline explanation!